src/main/cms/cms.c

```diff
@@ -990,7 +994,11 @@ void cmsUpdate(uint32_t currentTimeUs)
 
     if (!cmsInMenu) {
         // Detect menu invocation
+#ifdef USE_USB_CDC_HID
+        if (!usbCableIsInserted() && IS_MID(THROTTLE) && IS_LO(YAW) && IS_HI(PITCH) && !ARMING_FLAG(ARMED)) {
```

Comment:
Unfortunately this won't work reliably, since the majority of boards does not have an USB detect pin. In order to get this to work reliably, you'll have to check `getBatteryCellCount() == 0` as well.

In saying that, I don't think this is wanted behaviour for users, who are not using the HID facility, but want to do things like test OSD while connected over USB. Is there a way we can detect if the HID device is used, and use this to suppress the CMS?

Reply:
Huh I had no idea about that most boards don't use VBus. Ok in this case I'm pretty sure we can reliably detect usb presence if we tap into something in core files. Otherwise I have no good clue on what to do, some boards leak voltage which is backfeeded through dcdc converter. I made this mistake lol, so getBatteryCellCount may not be the best solution.

Reply:
Thats perfect :)

I think such check also need to be added to stick commands (e.g changing profiles/cam controll)

PS: Something looks wrong with your ssh/git key.. commit is from @khockuba and post here are from @conkerkh  ;)

Reply:
I know it's git config which is messed up:P

Reply:
@conkerkh: Since `getBatteryCellCount()` is used all over the place to check if a battery is present (or if we 'must be' on USB if not), the logic behind it is designed to be able to cope with USB voltage bleeding into the Vbat in: If Vbat is between `vbat_max` and `2 * vbat_min` (a range that typically includes 5V), the battery cell count will be reported as 0. But checking for USB HID activity is decidedly the better approach for this. I'd probably also check if `usb_hid_cdc` is enabled.

@wind0r: There were only 3 targets that had `USE_USB_DETECT` before we converted the targets that defined `VBUS_SENSING_PIN` to `USE_USB_DETECT`. The problem in this is that. since `VBUS_SENSING_PIN` wasn't actually used for anything, we don't know if it properly works on all of these targets. One potential pitfall is that if voltage from the 5V rail is bleeding into the USB Vcc pin, then the affected target will return true for `usbCableIsInserted()` as soon as a battery is connected. I'll open a pull request to display the result of `usbCableIsInserted()`  in `cli status`, to enable us to verify this for all of the targets with `USE_USB_DETECT`.